flow/e2e/clickhouse/clickhouse.go

```diff
@@ -0,0 +1,174 @@
+	return s.Connector().Conn()
+}
+
+func (s ClickHouseSuite) Suffix() string {
+	return s.suffix
+}
+
+func (s ClickHouseSuite) Peer() *protos.Peer {
+	return s.PeerForDatabase("e2e_test_" + s.suffix)
+}
+
+func (s ClickHouseSuite) PeerForDatabase(dbname string) *protos.Peer {
+	ret := &protos.Peer{
+		Name: e2e.AddSuffix(s, dbname),
+		Type: protos.DBType_CLICKHOUSE,
+		Config: &protos.Peer_ClickhouseConfig{
+			ClickhouseConfig: &protos.ClickhouseConfig{
+				Host:            "localhost",
+				Port:            9000,
+				Database:        dbname,
+				S3Path:          s.s3Helper.BucketName,
+				AccessKeyId:     *s.s3Helper.S3Config.AccessKeyId,
+				SecretAccessKey: *s.s3Helper.S3Config.SecretAccessKey,
+				Region:          *s.s3Helper.S3Config.Region,
+				DisableTls:      true,
+				Endpoint:        s.s3Helper.S3Config.Endpoint,
+			},
+		},
+	}
+	e2e.CreatePeer(s.t, ret)
+	return ret
+}
+
+func (s ClickHouseSuite) DestinationTable(table string) string {
+	return table
+}
+
+func (s ClickHouseSuite) Teardown() {
+	require.NoError(s.t, s.s3Helper.CleanUp(context.Background()))
+	e2e.TearDownPostgres(s)
+}
+
+func (s ClickHouseSuite) GetRows(table string, cols string) (*model.QRecordBatch, error) {
+	ch, err := connclickhouse.Connect(context.Background(), s.Peer().GetClickhouseConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	rows, err := ch.Query(
+		fmt.Sprintf(`SELECT %s FROM e2e_test_%s.%s ORDER BY id`, cols, s.suffix, table),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	batch := &model.QRecordBatch{}
+	types, err := rows.ColumnTypes()
+	if err != nil {
+		return nil, err
+	}
+	row := make([]interface{}, 0, len(types))
+	for _, ty := range types {
+		prec, scale, _ := ty.DecimalSize()
+		nullable, _ := ty.Nullable()
+		var qkind qvalue.QValueKind
+		switch ty.DatabaseTypeName() {
+		case "String":
+			var val string
+			row = append(row, &val)
+			qkind = qvalue.QValueKindString
+		case "Int32":
+			var val int32
+			row = append(row, &val)
+			qkind = qvalue.QValueKindInt32
+		default:
+			return nil, fmt.Errorf("failed to resolve QValueKind for %s", ty.DatabaseTypeName())
+		}
+		batch.Schema.Fields = append(batch.Schema.Fields, qvalue.QField{
+			Name:      ty.Name(),
+			Type:      qkind,
+			Precision: int16(prec),
+			Scale:     int16(scale),
+			Nullable:  nullable,
+		})
+	}
+
+	for rows.Next() {
+		if err := rows.Scan(row...); err != nil {
+			return nil, err
+		}
+		qrow := make([]qvalue.QValue, 0, len(row))
+		for _, val := range row {
+			switch v := val.(type) {
+			case *string:
+				qrow = append(qrow, qvalue.QValueString{Val: *v})
+			case *int32:
+				qrow = append(qrow, qvalue.QValueInt32{Val: *v})
+			default:
+				return nil, fmt.Errorf("cannot convert %T to qvalue", v)
+			}
+		}
+		batch.Records = append(batch.Records, qrow)
+	}
+
+	return batch, rows.Err()
+}
+
+func SetupSuite(t *testing.T) ClickHouseSuite {
+	t.Helper()
+
+	suffix := "ch_" + strings.ToLower(shared.RandomString(8))
```

Comment:
do we want to add timestamp here for future cleanup jobs?

Reply:
database running in action, no cleanup necessary